qiskit/assembler/assemble_schedules.py

```diff
@@ -80,8 +87,8 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
         })
 
     # setup pulse_library
-    qobj_config['pulse_library'] = [PulseLibraryItem(name=pulse.name, samples=pulse.samples)
-                                    for pulse in user_pulselib]
+    qobj_config['pulse_library'] = [PulseLibraryItem(name=name, samples=samples)
```

Comment:
If the above change is made to the pulse lib dictionary representation, this change would no longer be required.